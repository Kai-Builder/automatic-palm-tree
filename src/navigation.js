/* eslint-disable */
import AppBar from '@material-ui/core/AppBar';
import Home from '@material-ui/icons/Home';
import {Link} from 'react-router-dom';
import React, {Component} from 'react';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';

export class Navigation extends Component {

  constructor(props) {
    super(props);

    this.state = {
      value: 0
    }

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event, value) {
    this.setState({value});
  };

  render() {
    const {value} = this.state;

    return (
      <div className={this.props.className}>
        <AppBar position="static">
          <Tabs

            onChange={this.handleChange}
            value={value}
          >
            <Tab
              disableRipple={true}
              disableTouchRipple={true}
              component={Link}
              icon={<Home />}
              to='/'
            />
            <Tab
              disableRipple={true}
              disableTouchRipple={true}
              component={Link}
              label='Anonymous'
              to='/anonymous'
            />
            <Tab
              disableRipple={true}
              disableTouchRipple={true}
              component={Link}
              label='Restricted A'
              to='/restricted-a'
            />
            <Tab
              disableRipple={true}
              disableTouchRipple={true}
              component={Link}
              label='Restricted B'
              to='/restricted-b'
            />
          </Tabs>
        </AppBar>
      </div>
    )
  }
}
