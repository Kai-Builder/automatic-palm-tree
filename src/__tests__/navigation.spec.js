import {Navigation} from '../navigation';
import React from 'react';
import {shallow} from 'enzyme';

describe('Navigation', () => {
  const requiredProps = () => ({
  });
  const render = (props = requiredProps()) => shallow(<Navigation {...props}/>);

  test('component renders', () => {
    const wrapper = render();

    expect(wrapper).toMatchSnapshot();
  });
});
