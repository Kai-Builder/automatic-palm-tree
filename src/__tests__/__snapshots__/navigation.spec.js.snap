// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders 1`] = `
<div>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Tabs)
      onChange={[Function]}
      value={0}
    >
      <WithStyles(Tab)
        component={[Function]}
        disableRipple={true}
        disableTouchRipple={true}
        icon={<pure(HomeIcon) />}
        to="/"
      />
      <WithStyles(Tab)
        component={[Function]}
        disableRipple={true}
        disableTouchRipple={true}
        label="Anonymous"
        to="/anonymous"
      />
      <WithStyles(Tab)
        component={[Function]}
        disableRipple={true}
        disableTouchRipple={true}
        label="Restricted A"
        to="/restricted-a"
      />
      <WithStyles(Tab)
        component={[Function]}
        disableRipple={true}
        disableTouchRipple={true}
        label="Restricted B"
        to="/restricted-b"
      />
    </WithStyles(Tabs)>
  </WithStyles(AppBar)>
</div>
`;
